lib/Doctrine/ORM/Cache/DefaultCollectionHydrator.php

```diff
@@ -81,15 +81,34 @@ public function loadCacheEntry(ClassMetadata $metadata, CollectionCacheKey $key,
         $targetRegion    = $targetPersister->getCacheRegion();
         $list            = array();
 
+        $keys = array();
         foreach ($entry->identifiers as $index => $identifier) {
+            $keys[$index] = new EntityCacheKey($assoc['targetEntity'], $identifier);
+        }
+
 
-            $entityEntry = $targetRegion->get(new EntityCacheKey($assoc['targetEntity'], $identifier));
+        if ($targetRegion instanceof MultiGetRegion) {
```

Comment:
Can the region interface be changed to always accept multiple requested entries instead? That would make the multi-get at cache level much simpler, no?